canine/orchestrator.py

```diff
@@ -390,15 +409,24 @@ def make_output_DF(self, batch_id, outputs, cpu_time, prev_acct, localizer = Non
         except:
             df = pd.DataFrame()
 
+        # concatenate with any previously existing job avoided results
+        if self.df_avoided is not None:
+            df = pd.concat([df, self.df_avoided]).sort_index()
+
+        # save DF to disk
         if isinstance(localizer, AbstractLocalizer):
-            fname = "results.k9df.pickle"
+            fname = str(uuid.uuid4()) + "-results.k9df.pickle"
             df.to_pickle(fname)
             localizer.localize_file(fname, localizer.reserve_path(localizer.staging_dir, "results.k9df.pickle"))
             os.remove(fname)
```

Comment:
This has unexpected results in the BatchedLocalizer and it looks like you also have some special handling in the NFSLocalizer. Why don't we just sub this out for some localizer and backend-agnostic code:
```
            with localizer.transport_context() as transport:
                dest = localizer.reserve_path("results.k9df.pickle").controllerpath
                if not transport.isdir(os.path.dirname(dest)):
                    transport.makedirs(os.path.dirname(dest))
                with transport.open(dest, 'wb') as w:
                    df.to_pickle(w, compression=None)
```

Reply:
That would work much better! It would also avoid creating the results dataframe in the CWD and then moving it to the output directory.